### ISSUE TITLE
Possible dynamic min/max on RangeSlider workaround using facet values?

### ISSUE THREAD

**Issue by anthonycook** [meilisearch/instant-meilisearch#926]
Currently it is not possible to dynamically set min/max on the RangeSlider widget, I believe this is due to Meilisearch not yet supporting min/max on stats. See [discussion](https://github.com/meilisearch/product/discussions/117)

Would it be possible to implement this functionality in instant-meilisearch by using facets? For example if I had a price field that I wanted to use the RangeSlider with, could instant-meilisearch make a facet request for price and use the first and last values for min-max?

```json
"facetDistribution": {
    "price": {
      "1": 2099,   <<<< Min value 1
      "10": 2063,
      "100": 1995,
      "200": 2099  <<<< Max value is 200
     }
  }
````

This of course would be a temporary fix until Meilisearch supports min/max/avg ect. The RangeSlider widget would be way more useful with non-static min/max values.

Not sure if this is viable or has any potential issues, would appreciate feedback.

**Comment by gmourier** [comment]
Hi @anthonycook and @bidoubiwa 👋

This is a feature request (handling max/min) we are tracking [here](https://github.com/meilisearch/product/discussions/117)

As @bidoubiwa explains, the workaround will work to some extent. Ideally, we should be able to do it natively from Meilisearch.

Knowing the limitations expressed by @bidoubiwa, would it still be useful for your use case @anthonycook?

Thanks!

**Comment by anthonycook** [comment]
Hi @gmourier 👋

I have done some basic testing as I was concerned that not being able to set `maxValuesPerFacet` on facet by facet basis would not be performant enough.

I have an e-commerce dataset with over 20,000 products, when performing a search query on all 20k+ products with 8 facets and all fields returned I'm seeing minimal difference in response time when setting `maxValuesPerFacet` to a large value.

One possible concern is that I set `maxValuesPerFacet` to a large number (e.g say 2000) and in the future my data exceeds that number for a facet, range/price filters would then show the incorrect max value. Currently my price filter returns over 1200 unique values so it's possible this could happen as my data grows.


In conclusion this workaround works for my use case until max/min support officially lands in Meilisearch. It may work for others but I'd be cautious about exceeding your `maxValuesPerFacet` setting and performance might become an issue on larger datasets or data that has many more facets, so testing would be required.

**Comment by bidoubiwa** [comment]
If we implement it, I suppose we can add a setting that the user has to manually enable to ensure they are aware of what they are doing. Which limits the bad surprise.

**Comment by anthonycook** [comment]
> If we implement it, I suppose we can add a setting that the user has to manually enable to ensure they are aware of what they are doing. Which limits the bad surprise.

I think that would work, as it stands the range slider/input widgets feel very limited without dynamic min/max. It also leads to way more empty hits/results scenarios because you're having to guess the min/max values.

I think implementing this outweighs the one edge case as long as people are aware.

**Comment by bidoubiwa** [comment]
Hey @anthonycook, I'll be working on this feature once we merge #915 into `main`. To avoid too many conflicts.

**Comment by bidoubiwa** [comment]
Hey @anthonycook 

It is implemented in this PR: #1046 
and once this PR is merged: #1030 and released on `npm` you'll be able to use rangeSlider properly.
Don't forget to update to Meilisearch v1.1.0 at that point.
It should be next Monday :)

**Comment by bidoubiwa** [comment]
Hey! It is released. Please update to the latest version of instant-meilisearch and update your Meilisearch to v1.1.0
